File: gaming（框架代码）/gaming-design/csharp-code/DesignNotFoundException.cs
using System;

namespace YYDesign {

    public class DesignNotFoundException : Exception {
        private string message;
        private object[] args;
        public DesignNotFoundException(string message, params object[] args) {
            this.message = message;
            this.args = args;
        }

        public override string ToString() {
            return $"{message} can not find design object using args {args}";
        }
    }
}

File: gaming（框架代码）/gaming-design/src/main/java/org/gaming/design/exception/DesignNotFoundException.java
/**
 * 
 */
package org.gaming.design.exception;

import java.util.Arrays;

/**
 * @author YY
 *
 */
public class DesignNotFoundException extends RuntimeException {
	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	public static int stackTraceDepth = 10;

	private String method;
	private Object[] args;
	
	public DesignNotFoundException(String method, Object... args) {
		this.method = method;
		this.args = args;
		
		StackTraceElement[] stackTrace = this.getStackTrace();
		StackTraceElement[] mineStackTrace = new StackTraceElement[Math.min(stackTraceDepth, stackTrace.length)];
		System.arraycopy(stackTrace, 0, mineStackTrace, 0, mineStackTrace.length);
		this.setStackTrace(mineStackTrace);
	}

	@Override
	public String getMessage() {
		return method + " with args " + Arrays.toString(args) + " can not find config";
	}
}
